### PR TITLE
Making json output format easier to read

### PR DESCRIPTION
**Pull request by SinaChavoshi**
Modifying the diagnose_me output format for --json flag to make it easier to read.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2792)

### CODE REVIEW

File: sdk/python/kfp/cli/diagnose_me_cli.py [hunk 1]
```diff
@@ -27,7 +28,7 @@ def diagnose_me():
 @click.option(
     '--namespace',
     type=Text,
-    help='Namespace to use for Kubernatees cluster. It will use environment default if not specified.'
+    help='Namespace to use for Kubernetes cluster. It will use environment default if not specified.'
```
Comment (numerology) [inline sdk/python/kfp/cli/diagnose_me_cli.py]: Can you clarify whether it is a default value specified in env variable or the `default` namespace?

File: sdk/python/kfp/cli/diagnose_me_cli.py [hunk 2]
```diff
@@ -27,7 +28,7 @@ def diagnose_me():
...
 )
 @click.pass_context
 def diagnose_me(ctx, json, project_id, namespace):
```
Comment (numerology) [inline sdk/python/kfp/cli/diagnose_me_cli.py]: nit: `json` shadows outer package name `json`.

File: sdk/python/kfp/cli/diagnose_me_cli.py [hunk 3]
```diff
@@ -45,38 +46,58 @@ def diagnose_me(ctx, json, project_id, namespace):
...
+
+      if val.has_error:
+        print('Following error occurred during the diagnoses:', val.stderr)
+        continue
+
+      print(val.parsed_output)
+
   else:
-    print(results.json_output)
+    outptu_dict = {}
```
Comment (numerology) [inline sdk/python/kfp/cli/diagnose_me_cli.py]:
```suggestion
    output_dict = {}
```

File: sdk/python/kfp/cli/diagnose_me_cli.py [hunk 4]
```diff
@@ -45,38 +46,58 @@ def diagnose_me(ctx, json, project_id, namespace):
...
+    for key, val in results.items():
+
+      if val.has_error:
+        outptu_dict[
```
Comment (numerology) [inline sdk/python/kfp/cli/diagnose_me_cli.py]: nit: ditto

File: sdk/python/kfp/cli/diagnose_me_cli.py [hunk 5]
```diff
@@ -45,38 +46,58 @@ def diagnose_me(ctx, json, project_id, namespace):
...
+            key.name] = 'Following error occurred during the diagnoses: %s' % (
+                val.stderr)
+        continue
+      outptu_dict[key.name] = val.json_output
+
+    # Output results in Json format with indentation to make easy to read
+    print(
+        json.dumps(
+            outptu_dict, sort_keys=True, indent=2, separators=(',', ': ')))
```
Comment (numerology) [inline sdk/python/kfp/cli/diagnose_me_cli.py]: ditto

File: sdk/python/kfp/cli/diagnose_me_cli.py (hunk 4 above)
Comment (SinaChavoshi) [reply sdk/python/kfp/cli/diagnose_me_cli.py]: Done

File: sdk/python/kfp/cli/diagnose_me_cli.py (hunk 5 above)
Comment (SinaChavoshi) [reply sdk/python/kfp/cli/diagnose_me_cli.py]: done

File: sdk/python/kfp/cli/diagnose_me_cli.py (hunk 2 above)
Comment (SinaChavoshi) [reply sdk/python/kfp/cli/diagnose_me_cli.py]: good call! thank you.

File: sdk/python/kfp/cli/diagnose_me_cli.py (hunk 1 above)
Comment (SinaChavoshi) [reply sdk/python/kfp/cli/diagnose_me_cli.py]: The help message is in correct, if not specified --all-namespaces will be used. updated accordingly.

**Pull request merged**